enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/CausalClusterConfigurationValidator.java

```diff
@@ -38,41 +36,36 @@
 public class CausalClusterConfigurationValidator implements ConfigurationValidator
 {
     @Override
-    @Nonnull
-    public Map<String,String> validate( @Nonnull Collection<SettingValidator> settingValidators,
-            @Nonnull Map<String,String> rawConfig, @Nonnull Log log, boolean parsingFile ) throws InvalidSettingException
+    public Map<String,String> validate( @Nonnull Config config, @Nonnull Log log ) throws InvalidSettingException
     {
         // Make sure mode is CC
-        Mode mode = ClusterSettings.mode.apply( rawConfig::get );
-        if ( !mode.equals( Mode.CORE ) && !mode.equals( Mode.READ_REPLICA ) )
+        Mode mode = config.get( ClusterSettings.mode );
+        if ( mode.equals( Mode.CORE ) || mode.equals( Mode.READ_REPLICA ) )
         {
-            // Nothing to validate
-            return rawConfig;
+            validateInitialDiscoveryMembers( config );
+            validateBoltConnector( config );
+            validateLoadBalancing( config, log );
         }
 
-        validateInitialDiscoveryMembers( rawConfig::get );
-        validateBoltConnector( rawConfig );
-        validateLoadBalancing( rawConfig, log );
-
-        return rawConfig;
+        return Collections.emptyMap();
```

Comment:
Config validators are allowed to produce new values for the config. This behavior is questionable and should probably be removed in the future. We'll leave it for now since public API restricts refactoring.